internal/build/kubernetes.go

```diff
@@ -33,8 +33,16 @@ import (
 //go:embed templates/k8s_res_deploy.sh.tpl
 var k8sResDeployScriptTpl string
 
-func needsManifestsSetup(k *kubernetes.Kubernetes) bool {
-	return len(k.RemoteManifests) > 0 || len(k.LocalManifests) > 0
+func needsManifestsSetup(def *image.Definition) bool {
+	return len(def.Kubernetes.RemoteManifests) > 0 || len(def.Kubernetes.LocalManifests) > 0
+}
+
+func needsHelmChartsSetup(def *image.Definition) bool {
+	return (len(def.Release.Components.Helm) > 0) || def.Kubernetes.Helm != nil
+}
+
+func isKubernetesEnabled(def *image.Definition) bool {
+	return needsHelmChartsSetup(def) || needsManifestsSetup(def)
```

Comment:
Reading this I guess it's not possible to deploy an empty rke2 cluster?

Should we treat rke2 as it's own component to be able to deploy a box with "just" linux+kubernetes  and no workloads?

Reply:
Currently not possible, no.

The issue with that is there are already too many juggling parts and it feels too early to impose restrictions, e.g. if we have an entry for it in `release.yaml` or it is a separate flag, then the omission of this but inclusion of a Kubernetes manifest or a Helm chart will either have to fail (which is an inconvenient UX) or implicitly pass (which basically means the entry / flag is not respected at all).

Reply:
This was discussed offline and we decided to proceed with the current changes. Future iterations will improve on extensions and "empty" Kubernetes enablement.